### PR TITLE
Send Institution Collection Camp Dispatch Email

### PR DESCRIPTION
**Pull request by belwalshubham**
### Description
1- Send the dispatch email to the institution's POC and outcome email to coordinating POC if it is self-managed. If it is not self-managed, send the dispatch email and along with outcome to the camp attended by.
2. Send the ack. email to mmt once user fill the dispatch form.

## Summary by CodeRabbit

- **New Features**
	- Introduced functionality for sending logistics emails related to collection camps, including detailed error handling and email content generation.
	- Added a new API endpoint for handling cron jobs associated with institution collection camps, enabling automated logistics email dispatch.
	- Implemented notifications to the Material Management Team upon the creation of new entities.
	- Expanded tab functionality to include a new 'campOutcome' tab.

- **Bug Fixes**
	- Improved error logging during the logistics email sending process to enhance debugging capabilities.

### CODE REVIEW

File: wp-content/civi-extensions/goonjcustom/Civi/InstitutionCollectionCampService.php [hunk 1]
```diff
@@ -74,6 +78,127 @@ private static function generateInstitutionCollectionCampQrCode($id) {
...
+    if (empty($filteredItems)) {
+      return FALSE;
+    }
+
+    $goonjOfficeId = CustomField::get(FALSE)
+      ->addSelect('id')
+      ->addWhere('custom_group_id:name', '=', 'Camp_Vehicle_Dispatch')
+      ->addWhere('name', '=', 'To_which_PU_Center_material_is_being_sent')
+      ->execute()
+      ->first();
```
Comment (tarunnjoshi) [inline wp-content/civi-extensions/goonjcustom/Civi/InstitutionCollectionCampService.php:96]:
```suggestion
      ->execute()->first();
```

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/InstitutionCollectionCampCron.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+ */
+
+use Civi\Api4\EckEntity;
+use Civi\Api4\OptionValue;
+use Civi\InstitutionCollectionCampService;
+
+/**
+ * Goonjcustom.CollectionCampCron API specification (optional)
+ * This is used for documentation and validation.
+ *
```
Comment (tarunnjoshi) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/InstitutionCollectionCampCron.php:14]: It's not an `CollectionCampCron` when you are copying template from another file please update the name too

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/InstitutionCollectionCampCron.php [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+ * @throws \CRM_Core_Exception
+ */
+function civicrm_api3_goonjcustom_institution_collection_camp_cron($params) {
+  $returnValues = [];
+  $optionValues = OptionValue::get(FALSE)
+    ->addWhere('option_group_id:name', '=', 'eck_sub_types')
+    ->addWhere('name', '=', 'Institution_Collection_Camp')
+    ->addWhere('grouping', '=', 'Collection_Camp')
+    ->setLimit(1)
```
Comment (tarunnjoshi) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/InstitutionCollectionCampCron.php:42]: Why we need this

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/InstitutionCollectionCampCron.php [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    ->execute()->single();
+  $collectionCampSubtype = $optionValues['value'];
```
Comment (tarunnjoshi) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/InstitutionCollectionCampCron.php:44]: Put one space above it

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/InstitutionCollectionCampCron.php [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  $today = new DateTimeImmutable();
+  $endOfDay = $today->setTime(23, 59, 59)->format('Y-m-d H:i:s');
+  error_log("endOfDay: " . print_r($endOfDay, TRUE));
+  error_log("endOfDay: " . print_r($endOfDay, TRUE));
```
Comment (tarunnjoshi) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/InstitutionCollectionCampCron.php]: remove logs

**Review by tarunnjoshi — APPROVED** [review]

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/InstitutionCollectionCampCron.php (hunk 3 above)
Comment (belwalshubham) [reply wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/InstitutionCollectionCampCron.php:42]: To fetch the subtype of collection camp , which is `Institution_Collection_Camp` in my case

**Pull request merged**